### PR TITLE
17.03 cherry-picks

### PR DESCRIPTION
**Pull request by vieux**
image/cache: fix isValidParent logic #31189
Prevent freeing a possible invalid pointer from journald #31231
Synchronize the cursor returned by followJournal #31263
Fix duplicate mount points for multiple `--volumes-from` in `docker run` #29563
Shutdown leaks an error when the container was never started #31279

### CODE REVIEW

**Comment by thaJeztah** [comment]
@vieux CI is failing

```
22:14:11 # github.com/docker/docker/integration-cli
22:14:11 ./docker_cli_build_test.go:7091: too many arguments in call to fakeContext
22:14:11 ./docker_cli_build_test.go:7091: multiple-value fakeContext() in single-value context
22:14:11 ./docker_cli_build_test.go:7094: undefined: buildImageSuccessfully
22:14:11 ./docker_cli_build_test.go:7094: undefined: withExternalBuildContext
22:14:11 ./docker_cli_build_test.go:7095: too many arguments in call to getIDByName
22:14:11 ./docker_cli_build_test.go:7095: multiple-value getIDByName() in single-value context
22:14:11 ./docker_cli_build_test.go:7098: undefined: withBuildFlags
22:14:11 ./docker_cli_build_test.go:7098: undefined: withExternalBuildContext
22:14:11 ./docker_cli_build_test.go:7098: multiple-value buildImage() in single-value context
22:14:11 ./docker_cli_build_test.go:7100: too many arguments in call to getIDByName
22:14:11 ./docker_cli_build_test.go:7100: too many errors
```

File: CHANGELOG.md [hunk 1]
```diff
@@ -24,6 +25,10 @@ be found.
 
 * Fix a deadlock in docker logs [#30223](https://github.com/docker/docker/pull/30223)
 * Fix cpu spin waiting for log write events [#31070](https://github.com/docker/docker/pull/31070)
+* Fix a possible crash when using journald [#31231](https://github.com/docker/docker/pull/31231) [#31263](https://github.com/docker/docker/pull/31231)
+* Fix a panic onclose of nil channel [#31274](https://github.com/docker/docker/pull/31274)
```
Comment (thaJeztah) [inline CHANGELOG.md]: on close

**Review by thaJeztah — APPROVED** [review]
LGTM

**Review by vdemeester — APPROVED** [review]
LGTM 🦁

**Pull request merged**